### PR TITLE
Joshua/futr 955 update workflow docs

### PR DESCRIPTION
**Pull request by Joshuafolorunsho**
Pizza Workflow Example Documentation
- Replaced web crawler example with relatable pizza ordering workflow
- Added clear step-by-step guide for creating workflows
- Improved code examples with proper TypeScript formatting
- Demonstrated key workflow concepts (validation, dependencies, conditions)
- Restructured documentation for better readability
- Fixed file paths and code style for consistency

### CODE REVIEW

**Comment by Joshuafolorunsho** [comment]
```
@vercel
Vercel – openapi-spec-writer — Deployment has failed
```

Openapi spec writer breaking doesn't have anything to do with docs changes.

File: docs/src/pages/guide/how-to/03-building-workflows.mdx [hunk 1]
```diff
@@ -1,131 +1,171 @@
...
+
+```ts filename="src/mastra/workflows/index.ts" showLineNumbers copy
+// Map trigger data to first step
+pizzaWorkflow.config("validatePizzaOrder", {
+  variables: {
+    size: { stepId: "trigger", path: "size" },
+    toppings: { stepId: "trigger", path: "toppings" },
+    specialInstructions: { stepId: "trigger", path: "specialInstructions" },
+  },
+});
```
Comment (adeniyii) [inline docs/src/pages/guide/how-to/03-building-workflows.mdx:117]: we need a `dependsOn` property here
```suggestion
pizzaWorkflow.config("validatePizzaOrder", {
  dependsOn: [],
  variables: {
    size: { stepId: "trigger", path: "size" },
    toppings: { stepId: "trigger", path: "toppings" },
    specialInstructions: { stepId: "trigger", path: "specialInstructions" },
  },
});
```

File: docs/src/pages/guide/reference/workflows/config.mdx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+- `condition`: A simple boolean expression to determine if the step should run.
+
+  ```ts copy
+  workflow.config("sendRefund", {
+    condition: {
+      ref: { stepId: "validateRefund", path: "result.isEligible" },
+      query: { $eq: true },
+    },
+  });
+  ```
```
Comment (adeniyii) [inline docs/src/pages/guide/reference/workflows/config.mdx:35]: here as well

File: docs/src/pages/guide/reference/workflows/config.mdx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+- `variables`: Defines how to map data from previous steps into the current step's input.
+
+  ```ts copy
+  workflow.config("sendEmail", {
+    variables: {
+      userId: { stepId: "validateUser", path: "result.userId" },
+      orderData: { stepId: "processOrder", path: "result" },
+    },
+  });
+  ```
```
Comment (adeniyii) [inline docs/src/pages/guide/reference/workflows/config.mdx:24]: here too

File: docs/src/pages/guide/reference/workflows/config.mdx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+- `conditionFn`: A custom function that returns a boolean to determine if the step should run. Useful for complex conditions.
+  ```ts copy
+  workflow.config("sendNotification", {
+    conditionFn: async ({ steps }) => {
+      const orderValue = steps.processOrder.output.total;
+      const userType = steps.validateUser.output.type;
+      return orderValue > 1000 && userType === "premium";
+    },
+  });
+  ```
```
Comment (adeniyii) [inline docs/src/pages/guide/reference/workflows/config.mdx:46]: here

File: docs/src/pages/guide/reference/workflows/workflow.mdx [hunk 5]
```diff
@@ -90,13 +116,13 @@ Variables payloads are typesafe with fields defined in the [Step](./step.mdx) `i
...
     // Access trigger data
     variables: {
       orderId: { stepId: "trigger", path: "orderId" },
     },
   })
-  .step("processPayment", {
+  .config("processPayment", {
     variables: {
       // Access previous step's data
       orderStatus: { stepId: "createOrder", path: "status" },
```
Comment (adeniyii) [inline docs/src/pages/guide/reference/workflows/workflow.mdx:128]: and here

**Review by adeniyii — CHANGES_REQUESTED** [review]
small errors

**Comment by calcsam** [comment]
Thx for closing. Once we fix the APIs, I will re-do the docs.

**Comment by Joshuafolorunsho** [comment]
> Thx for closing. Once we fix the APIs, I will re-do the docs.

You're welcome.

**Pull request closed without merging**